tests/conftest.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+import pytest
+from shutil import copytree, rmtree
+
+
+def pytest_addoption(parser):
+    parser.addoption('--quick', action='store_true', help='Use existing input '
+                                                              'files in tests/data')
+
+
+@pytest.fixture(scope='session')
+def quick_run(pytestconfig):
+    return pytestconfig.option.quick
+
+
+@pytest.fixture(scope='session')
+def setup_input_files(quick_run):
+    if not quick_run:
```

Comment:
I feel like this requires a least a little bit of explanation. I think I understand what pytest magic is happening behind the scenes, but it _looks_ like `quick_run` is both a function and a boolean, which would be an unfortunately confusing case of variable name shadowing. But I understand it might be the case that this naming is required because the names of things affect pytest fixtures.

Reply:
Fixtures can return data to the test function, and this is what is happening here with the boolean. The name of the parameter `quick_run` has to match the name of the fixture.

Reply:
Fun fact: it doesn't *absolutely* have to! :-)  If you, like me, are bothered by this, you can decorate the quick_run function with `@pytest.fixture(name='is_it_a_quick_run')`, and then call this as

```python
def setup_input_files(is_it_a_quick_run):
    if not is_it_a_quick_run:
```
(disclaimer: you should pick a better name than that.)